samples/boards/96b_argonkey/microphone/src/main.c

```diff
@@ -0,0 +1,224 @@
+
+K_MEM_SLAB_DEFINE(rx_mem_slab, BLOCK_SIZE_BYTES, NUM_RX_BLOCKS, 1);
+K_MEM_SLAB_DEFINE(tx_mem_slab, BLOCK_SIZE_BYTES, NUM_TX_BLOCKS, 1);
+
+#define NUM_MS		3000
+u16_t pcm_out[32*NUM_MS];
+#endif
+
+#define NUM_LEDS 12
+#define DELAY_TIME K_MSEC(25)
+
+void signal_sampling_started(void)
+{
+	static struct device *led0, *led1;
+
+	led0 = device_get_binding(LED0_GPIO_CONTROLLER);
+	gpio_pin_configure(led0, LED0_GPIO_PIN, GPIO_DIR_OUT);
+	gpio_pin_write(led0, LED0_GPIO_PIN, 1);
+
+	led1 = device_get_binding(LED1_GPIO_CONTROLLER);
+	gpio_pin_configure(led1, LED1_GPIO_PIN, GPIO_DIR_OUT);
+	gpio_pin_write(led1, LED1_GPIO_PIN, 0);
+}
+
+void signal_sampling_stopped(void)
+{
+	static struct device *led0, *led1;
+
+	led0 = device_get_binding(LED0_GPIO_CONTROLLER);
+	gpio_pin_configure(led0, LED0_GPIO_PIN, GPIO_DIR_OUT);
+	gpio_pin_write(led0, LED0_GPIO_PIN, 1);
+
+	led1 = device_get_binding(LED1_GPIO_CONTROLLER);
+	gpio_pin_configure(led1, LED1_GPIO_PIN, GPIO_DIR_OUT);
+	gpio_pin_write(led1, LED1_GPIO_PIN, 1);
+}
+
+void signal_print_stopped(void)
+{
+	static struct device *led0, *led1;
+
+	led0 = device_get_binding(LED0_GPIO_CONTROLLER);
+	gpio_pin_configure(led0, LED0_GPIO_PIN, GPIO_DIR_OUT);
+	gpio_pin_write(led0, LED0_GPIO_PIN, 0);
+
+	led1 = device_get_binding(LED1_GPIO_CONTROLLER);
+	gpio_pin_configure(led1, LED1_GPIO_PIN, GPIO_DIR_OUT);
+	gpio_pin_write(led1, LED1_GPIO_PIN, 1);
+}
+
+void main(void)
+{
+	int i;
+	u32_t ms_count;
+
+#ifdef CONFIG_LP3943
+	static struct device *ledc;
+
+	ledc = device_get_binding(CONFIG_LP3943_DEV_NAME);
+	if (!ledc) {
+		printk("Could not get pointer to %s sensor\n",
+			CONFIG_LP3943_DEV_NAME);
+		return;
+	}
+
+	/* turn all leds on */
+	for (i = 0; i < NUM_LEDS; i++) {
+		led_on(ledc, i);
+		k_sleep(DELAY_TIME);
+	}
+
+	/* turn all leds off */
+	for (i = 0; i < NUM_LEDS; i++) {
+		led_off(ledc, i);
+		k_sleep(DELAY_TIME);
+	}
+
+#endif
+
+	printk("ArgonKey test!!\n");
+
+#ifdef CONFIG_I2S
+	struct i2s_config i2s_cfg;
+	int a_ret;
+	void *rx_block;
+	size_t rx_size;
+
+	struct device *adev = device_get_binding(CONFIG_I2S_5_NAME);
+
+	if (!adev) {
+		printk("Could not get pointer to %s sensor\n",
+			CONFIG_I2S_5_NAME);
+		return;
+	}
+
+	i2s_cfg.word_size = CHAN_SIZE;
+	i2s_cfg.channels = 1;
+	i2s_cfg.format = I2S_FMT_DATA_FORMAT_LEFT_JUSTIFIED |
+			 I2S_FMT_BIT_CLK_INV;
+	i2s_cfg.options = I2S_OPT_FRAME_CLK_MASTER | I2S_OPT_BIT_CLK_MASTER;
+	i2s_cfg.frame_clk_freq = AUDIO_FREQ * OVERSAMPLING_FACTOR / CHAN_SIZE;
+	i2s_cfg.block_size = BLOCK_SIZE * sizeof(s16_t);
+	i2s_cfg.mem_slab = &rx_mem_slab;
+	i2s_cfg.timeout = 2000;
+
+	/* init the filter lib */
+	struct TPDMFilter filter_init;
+
+	filter_init.LP_HZ = AUDIO_FREQ / 2;
+	filter_init.HP_HZ = 10;
+	filter_init.Fs = AUDIO_FREQ;
+	filter_init.Out_MicChannels = 1;
+	filter_init.In_MicChannels = 1;
+	filter_init.Decimation = OVERSAMPLING_FACTOR;
+	filter_init.MaxVolume = 64;
+
+	Open_PDM_Filter_Init(&filter_init);
```

Comment:
Access should be done through a driver.

Reply:
But I agree to move the OpenPDMFilter library in the ext/hal/st/lib directory to be used by
everyone. Please not that this library is the one with following license:

http://www.st.com/software_license_agreement_image_v2

Reply:
I think that a empty driver would be required just for the library license, it is provided to work only with MP34DT05, it could and should be enforced by the use of a dedicated driver.

Reply:
OK, I'm leaving for the day. I'm resubmitting a new PR with everything addresses by you and dbkinder except this stuff. I will work on it tomorrow morning.